src/Modules/ConfigOptions.lua

```diff
@@ -1622,77 +1638,64 @@ Uber Pinnacle Boss adds the following modifiers:
-			build.configTab.varControls['enemyCritChance']:SetPlaceholder(0, true)
-		elseif val == "Shaper Ball" then
-			if build.configTab.enemyLevel then
-				build.configTab.varControls['enemyColdDamage']:SetPlaceholder(round(data.monsterDamageTable[build.configTab.enemyLevel] * data.bossSkills["Shaper Ball"].damageMult), true)
+			
+			if bossData.DamageType then
+				build.configTab.varControls['enemyDamageType']:SelByValue(bossData.DamageType, "val")
+				build.configTab.input['enemyDamageType'] = bossData.DamageType
 			end
-
-			build.configTab.varControls['enemyColdPen']:SetPlaceholder(25, true)
-			build.configTab.varControls['enemySpeed']:SetPlaceholder(data.bossSkills["Shaper Ball"].speed, true)
 			build.configTab.varControls['enemyDamageType'].enabled = false
-			build.configTab.varControls['enemyDamageType']:SelByValue("SpellProjectile", "val")
-			build.configTab.input['enemyDamageType'] = "SpellProjectile"
-		elseif val == "Shaper Slam" then
-			if build.configTab.enemyLevel then
-				build.configTab.varControls['enemyPhysicalDamage']:SetPlaceholder(round(data.monsterDamageTable[build.configTab.enemyLevel] * data.bossSkills["Shaper Slam"].damageMult), true)
+			
+			if bossData.speed then
+				build.configTab.varControls['enemySpeed']:SetPlaceholder(bossData.speed, true)
 			end
-			build.configTab.varControls['enemyDamageType'].enabled = false
-			build.configTab.varControls['enemyDamageType']:SelByValue("Melee", "val")
-			build.configTab.input['enemyDamageType'] = "Melee"
-
-			build.configTab.varControls['enemySpeed']:SetPlaceholder(data.bossSkills["Shaper Slam"].speed, true)
-		elseif val == "Maven Memory Game" then
-			if build.configTab.enemyLevel then
-				local defaultEleDamage = round(data.monsterDamageTable[build.configTab.enemyLevel] * data.bossSkills["Maven Memory Game"].damageMult)
-				build.configTab.varControls['enemyLightningDamage']:SetPlaceholder(defaultEleDamage, true)
-				build.configTab.varControls['enemyColdDamage']:SetPlaceholder(defaultEleDamage, true)
-				build.configTab.varControls['enemyFireDamage']:SetPlaceholder(defaultEleDamage, true)
+			if bossData.critChance then
+				build.configTab.varControls['enemyCritChance']:SetPlaceholder(bossData.critChance, true)
 			end
-			build.configTab.varControls['enemyDamageType'].enabled = false
-			build.configTab.varControls['enemyDamageType']:SelByValue("Melee", "val")
-			build.configTab.input['enemyDamageType'] = "Melee"
+			
+			modList:NewMod("BossSkillActive", "FLAG", true, "Config")
+		else
+			build.configTab.varControls['enemyDamageType'].enabled = true
 		end
 	end },
+	{ var = "enemyDamageRollRange", type = "integer", label = "Enemy Skill Roll Range %:", ifFlag = "BossSkillActive", tooltip = "The percentage of the roll range the enemy hits for \n eg at 100% the enemy deals its maximum damage", defaultPlaceholderState = 70, hideIfInvalid = true },
```

Comment:
This would be better as a slider but would require a bit more work to implement them in the config tab.